.github/workflows/links.yml

```diff
@@ -61,3 +61,17 @@ jobs:
         env:
           # to be used in case rate limits are surpassed
           GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+  linkcheckerSphinx:
+    runs-on: ubuntu-latest
+    needs: linkCheckerLychee
+    
+    steps:
+      # Checkout repository code
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      # Run Sphinx's linkcheck builder
+      - name: Run Sphinx linkcheck
+        run: |
+          sphinx-build -b linkcheck . _build/linkcheck || exit 1 #Fail the job if the link is broken 
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
The failure is probably coming from needing to install PyBaMM[all,docs] before running the check. I have no looked at the linkcheck option before, but you might also need to build the docs independently

Reply:
This might be something to include int the docs build in nox instead of here. Otherwise we will be doing 3 documentation builds

Reply:
Thanks @kratman ,  I will make sure documentation is built before the link check

Reply:
There are two options:
- we should use `scientific-python/cookie`'s pattern here: https://github.com/scientific-python/cookie/blob/95e0bb06ccf09163bed4e2c6d5726e5990c70a13/%7B%7Bcookiecutter.project_name%7D%7D/noxfile.py#L48-L75 and create a small argument parser that allows choosing between the HTML builder or the linkcheck builder
- or, just install PyBaMM here with `[docs,all]` without modifying the `nox` session before running `sphinx-build -b linkcheck`. The linkcheck builder calls the HTML builder internally

I slightly lean towards the second option, because I don't think anyone is going to use this session for local development.

I agree on not having redundant CI jobs, though; it'd probably be nice to move all docs-related jobs (docs build, doctests, and this linkchecker build) to a `docs.yml` workflow file. Our CI is not winning any timing competition, so they can all be steps that run in serial in a single job instead of separate jobs.

Reply:
@agriyakhetarpal  I went with the second option of installing PyBaMM with ```[docs,all]```  . Kindly review the new code for further modifications if needed. I also ran the pre-commit this time